src/crimson/os/seastore/transaction.h

```diff
@@ -141,8 +141,25 @@ class Transaction {
     return retired_set;
   }
 
+  enum class src_t : uint8_t {
+    SEASTORE_MUTATE = 0,
+    SEASTORE_READ,
+    CLEANER,
+    // including tests and tools that bypass seastore
+    TEST,
+    /**
+     * If set, *this may not be used to perform writes and will not provide
+     * consistentency allowing operations using to avoid maintaining a read_set.
+     */
+    WEAK,
```

Comment:
WEAK is orthogonal to source.  Remove from src_t and add the weak boolean and helpers back in.  Probably worth adding an INIT src_t value for mkfs/mount though.

Reply:
The current `SEASTORE_READ` is categorized as seastore-level read operations that expect conflicts. This is why we can calculate the accurate conflict rate from equation `trans_invalidated[SEASTORE_READ] / trans_created[SEASTORE_READ]`.

Strictly speaking we need another orthogonal label maybe called "op_type_t" including "READ", "WRITE", and "WEAK", and "src_t" would include "TEST", "SEASTORE", "CLEANER" and "INIT". But that would be too many and some combinations don't exist or not meaningful for performance profiling (mostly unit tests), that's why I combined them into the existing 5 categories.

Reply:
WEAK is irrelevant to profiling.  INIT would be the op type as it refers to mkfs and mount.  I want op_type_t to be purely informational and do not want it to impact behavior.  Please restore the weak boolean and related behavior.

Reply:
reverted the weak boolean in https://github.com/ceph/ceph/pull/42202/commits/312340317e1f8bda31a9b27ec4ea7c3396dcb088